### ISSUE TITLE
Coverflow versus "Cards"

### ISSUE THREAD

**Issue by politicalslug** [rpetrich/ProSwitcher#69]
This app has been executed flawlessly. This is just a beta (0.7) and I haven't encountered a single problem.

Now that I've said that: I do have a problem. It isn't with the coding, it's with the concept. I just don't understand the obsession with Palm's "cards" method of multitasking. Coverflow is a much better method, both aesthetically and functionally. Coverflow angles the apps on the sides so instead of seeing just a fragment of either app on the left or right, you can see a few on either side, because they are angled. While this works better in landscape than portrait, it's still doable. The "cards" method leaves me flicking more than necessary if I have more than three apps open. That's not cool at all. What really is need is two different views. Kirikae works better in portrait mode because everything can be seen at a glance with little flicking needed. That's great! In landscape mode this concept works better.

Here is my suggestion: employ (at least optionally) a Kirikae like interface for portrait (without previews, just icons and text) and then invoke Coverflow view for landscape mode (turning the phone on its side). Since coverflow only allows so much viewing area in portrait mode, another option could be to shrink each preview window, with the option to zoom in if you hover (hold for a second) on the preview. Additionally, perhaps a rubberband approach would be better than having to flick for each preview window (like a hard swipe sends you across multiple previews instead of just to the next in line).

<-- my two cents.

**Comment by grp** [comment]
Too bad I hate coverflow :P.

**Comment by politicalslug** [comment]
That's it?  You honestly think the best approach to multitasking is to show just one app at a time with only one/eighth of the next app showing (not enough to distinguish which app that might be as it's just a tiny sliver).  Couple that with having to swipe across each preview because only one can be seen at a time.  So if you have 10 apps running, and you need to get from the first to the last, you have to swipe nine times.  With coverflow that could be accomplished in just two short swipes and in about a third or less the time.  I don't believe for a second you actually believe that's a better approach.  Too much time is spent swiping.  That's just silly.

Are you just being obstinate or are you messing with me?

Also, how about the Landscape only mode, with either Kirikae allowed for Portrait mode, or you adding in a substitute with icons and text only?

**Comment by rpetrich** [comment]
Coverflow is not in our roadmap. The foundations of a cover-flow style are present in ProSwitcher though; perhaps someone will fork it and implement that feature

Note: cover-flow will not be simple to implement and it will adversely effect the performance/testability of our current design

**Comment by politicalslug** [comment]
Thanks for the response.  I understand the performance concerns, but if indeed the nextgen iPhone carriers a dual-core cpu, performance problems could be solved by running the coverflow rendering as a seperate process.  Just a thought.  Any ideas on a text/icon only view like Kirikae for portrait mode and implementing the 'card' approach either significantly zoomed out or only in landscape mode?

**Comment by rpetrich** [comment]
We are targeting all the way back to the original iPhone. 90% of the rendering is done on the GPU, so that is where the performance boost would be needed; the 3G S GPU does well, the older ones not so much.

We have no plans to reimplement kirikae (it's open-source and works really well so why would we) and again will accept landscape patches if they're of sufficient quality.

Glad you like LockLight, your comment has been removed due to language though :(

**Comment by politicalslug** [comment]
Sorry.  My bad.  No more foul language henceforth.

**Comment by GabsNL** [comment]
Even do CoverFlow would have been nice I hope this will not be implented.
My iPhone 3G isn't exactly a powerhouse and my mate bought a (even slower) first generation iPod last month. I love how how ProSwitcher runs very smooth, I wouldn't want to jeopardies this just for eye candy.

What I think could be done are your rubber band idea and perhaps the settings or ability to make the 'cards' zoom out more (like when reordering cards on the Palm Pre) to show 4 - 5 apps at the same time.

**Comment by politicalslug** [comment]
I would hate to make things more difficult for the developers, but I have to figure that those on contract with a subsidized iPhone (ie 3G or 3GS) are going to upgrade every two years. Surely forking development for newer devices (3GS, 3G iPod Touch and future devices which support openGL and advanced graphics rendering) and older devices (3G and 2G iPhones have identical hardware, as do iPod Touch 1G and 2G) makes some sense.  Let us not be constrained by the lowest common denominator.

**Comment by rpetrich** [comment]
If other developers want to maintain a "advanced graphics" fork they can do so. We have no plans to.

**Comment by ejoya** [comment]
I don't think majority of the users will have 10 backgrounded apps to use coverflow. It will definitely slow down the phone. Basically normal users will have around 4-7 apps at most on background

I think the developers can put a button or something to go to the last backgrounded app and button to go to the first backgrounded app. (Like scroll top and scroll down of actionmenu). This will reduce the swiping by half.

**Comment by politicalslug** [comment]
Good call.  Perhaps a setting for ordering apps by last used, or even better, most frequently used.  Maybe we can assign a priority tag to each of our apps.  Like high-medium-low and organize that way (so my GPS app might be a high, while the ipod could a medium, and the notepad maybe a low).  

Whatever.  For the time being I'm using this in conjunction with Kirikae (no issues whatsoever so far as I can tell) and proswitcher is on my dock (with dock showing) and Kirikae is invoked by the 2 second home button hold.  I find that proswitcher works faster from the home screen than from an app (but the difference is probably less than half of second on my 3GS).

**Comment by rpetrich** [comment]
Double-tap to the right/left of a screenshot to jump to the start/end of the list.
Reordering cards is on the to do list

**Comment by politicalslug** [comment]
Also, in addition to a button for instantly going to the back or front, unlocking (allowing rubberbanding) the preview windows so it scrolls until you stop it, instead of stopping on each window.  Thanks rpetrich and chpwn!  You guys are brilliant and I'm just happy to get the benefit of your minds in the form of iPhone apps that make my mobile experience more enjoyable.

**Comment by ejoya** [comment]
@rpetrich wow. I didn't know it's already implemented, this is great. Thanks again

**Comment by politicalslug** [comment]
Wow that was quick.  You're right.  Just tried it and that works well.  Buttons could work too though, or having both invocation methods.

**Comment by rpetrich** [comment]
Disabling paging is a simple solution that could be made into an option; I'll look into it.

I'm opposed to adding buttons; destroys the clean and simple design

**Comment by politicalslug** [comment]
Fair enough.  I suppose it would add unneeded glut to the screen.  That's would be great if you could add that as an option.

**Comment by rpetrich** [comment]
Added as hidden setting in unstable: PSWPagingEnabled

**Comment by ejoya** [comment]
sorry to bother but what do you mean by unstable? is there an unstable build other than 0.7.2? I would love to test it there's one

**Comment by politicalslug** [comment]
You'll have to forgive my ignorance, but does that mean I can use it now (if so, how) or that it will be implemented in a future build?

**Comment by politicalslug** [comment]
LOL. Unstable other than 0.7.2 (which is wholly stable).  That gave me a chuckle.

**Comment by GabsNL** [comment]
Unstable is the unstable branch of the source code.
You can compile it yourself from this site if you know how.
When unstable becomes stable it gets released in .deb package in the repo.

**Comment by politicalslug** [comment]
Ahhh.  Very good.  I'll wait for a better man to compile the code (as I don't know how to).  Thanks for answering all our questions so very quickly.

**Comment by ejoya** [comment]
@gabsnl ok thanks. I will wait for the stable release then. lol

**Comment by politicalslug** [comment]
This is a little off topic, but is there a way to activate the pop-up iPod controls by doing the 2second hold of the home or sleep button, so I can use the double tap method for invoking proswitcher?

**Comment by rpetrich** [comment]
Off topic; and no there is no way currently. Conversations are best suited for IRC/IM/Twitter, not a bug tracker

**Comment by GabsNL** [comment]
As rpetrich suggested you guys should come and discus it in the IRC.
add irc.saurik.com and join the room #ProSwitcher
